ilc/common/Slot/SpaSlot.js

```diff
@@ -0,0 +1,74 @@
+import {assert} from '../../client/utils';
+
+export class SpaSlot {
+    #applicationId;
+    #applicationName;
+    #slotName;
+    /**
+     * @property {IlcConfigRoot}
+     */
+    #configRoot
+
+    /**
+     * @constructor
+     * @param {{slotName: string, appName: string, appId: string}} slotName
+     * @param {IlcConfigRoot} rootConfig
+     */
+    constructor({applicationId, applicationName, slotName}, configRoot) {
+
+        assert(applicationId !== 'string', 'SpaSlot instance can not be initiated without applicationId');
+        assert(applicationName !== 'string', 'SpaSlot instance can not be initiated without applicationName');
+        assert(slotName !== 'string', 'SpaSlot instance can not be initiated without slotName');
+
+        this.#applicationId = applicationId;
+        this.#applicationName = applicationName;
+        this.#slotName = slotName;
+        this.#configRoot = configRoot;
+    }
+
+    /**
+     * @method
+     * @return {string} ApplicationName
+     */
+    getApplicationName() {
```

Comment:
can we use getters for it?

Reply:
We can. But I decided not to use it based on current code style. I would propose to switch to getter with transition to TS and setuping eslint into the project.